File: Common/Serialization/StringFormatterBase.vb
Public MustInherit Class StringFormatterBase
    Inherits Formatter

    Public Sub New()
        Me.Serializers.Add(Serializer(Of Boolean).Create(
                        $"Binary_{NameOf([Boolean])}",
                        Function(F) ParseInv.Boolean(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))

        Me.Serializers.Add(Serializer(Of SByte).Create(
                        $"Binary_{NameOf([SByte])}",
                        Function(F) ParseInv.SByte(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of Int16).Create(
                        $"Binary_{NameOf(Int16)}",
                        Function(F) ParseInv.Short(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of Int32).Create(
                        $"Binary_{NameOf(Int32)}",
                        Function(F) ParseInv.Integer(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of Int64).Create(
                        $"Binary_{NameOf(Int64)}",
                        Function(F) ParseInv.Long(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of Byte).Create(
                        $"Binary_{NameOf([Byte])}",
                        Function(F) ParseInv.Byte(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of UInt16).Create(
                        $"Binary_{NameOf(UInt16)}",
                        Function(F) ParseInv.UShort(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of UInt32).Create(
                        $"Binary_{NameOf(UInt32)}",
                        Function(F) ParseInv.UInteger(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of UInt64).Create(
                        $"Binary_{NameOf(UInt64)}",
                        Function(F) ParseInv.ULong(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))

        Me.Serializers.Add(Serializer(Of Single).Create(
                        $"Binary_{NameOf([Single])}",
                        Function(F) ParseInv.Single(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))
        Me.Serializers.Add(Serializer(Of Double).Create(
                        $"Binary_{NameOf([Double])}",
                        Function(F) ParseInv.Double(F.Get(Of String)(Nothing)),
                        Nothing,
                        Sub(F, O) F.Set(Nothing, O.ToStringInv())))

        Me.Serializers.Add(Serializer(Of Char).Create(
                        $"Binary_{NameOf([Char])}",
                        Function(F) F.Get(Of String)(Nothing).Chars(0),
                        Nothing,
                        Sub(F, O) F.Set(Of String)(Nothing, O)))

        Me.Initialize()
    End Sub

End Class

File: Common/Serialization/StringFormatterBase.cs
using System;

namespace Ks
{
    namespace Common
    {
        public abstract class StringFormatterBase : Formatter
        {
            public StringFormatterBase()
            {
                this.Serializers.Add(Serializer<bool>.Create($"Binary_{nameof(Boolean)}", F => ParseInv.Boolean(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));

                this.Serializers.Add(Serializer<sbyte>.Create($"Binary_{nameof(SByte)}", F => ParseInv.SByte(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<Int16>.Create($"Binary_{nameof(Int16)}", F => ParseInv.Short(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<Int32>.Create($"Binary_{nameof(Int32)}", F => ParseInv.Integer(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<Int64>.Create($"Binary_{nameof(Int64)}", F => ParseInv.Long(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<byte>.Create($"Binary_{nameof(Byte)}", F => ParseInv.Byte(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<UInt16>.Create($"Binary_{nameof(UInt16)}", F => ParseInv.UShort(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<UInt32>.Create($"Binary_{nameof(UInt32)}", F => ParseInv.UInteger(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<UInt64>.Create($"Binary_{nameof(UInt64)}", F => ParseInv.ULong(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));

                this.Serializers.Add(Serializer<float>.Create($"Binary_{nameof(Single)}", F => ParseInv.Single(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));
                this.Serializers.Add(Serializer<double>.Create($"Binary_{nameof(Double)}", F => ParseInv.Double(F.Get<string>(null)), null, (F, O) => F.Set(null, O.ToStringInv())));

                this.Serializers.Add(Serializer<char>.Create($"Binary_{nameof(Char)}", F => F.Get<string>(null)[0], null, (F, O) => F.Set<string>(null, O.ToString())));

                this.Initialize();
            }
        }
    }
}
